### PR TITLE
Region filter in planning areas

### PR DESCRIPTION
**Pull request by pflopez**
- Adds region filter on planning areas
- Updates url with region filters
- Fixes some styles on planning areas
- Fixes some details on filter-dropdown component

### CODE REVIEW

File: src/interface/src/app/types/region.types.ts [hunk 1]
```diff
@@ -27,3 +27,22 @@ export function regionToString(region: Region | null): string {
   }
   return 'sierra-nevada';
 }
+
+export const RegionsWithString = [
```
Comment (pflopez) [inline src/interface/src/app/types/region.types.ts:31]: not a fan of this, I think we need to rework how we deal with regions in general, but I dont want to bite that bullet on this pr

Comment (lastminutediorama) [reply src/interface/src/app/types/region.types.ts:31]: Fair point -- I'm also not sure how we should deal with these, especially if/when these expand.

Comment (pflopez) [reply src/interface/src/app/types/region.types.ts:31]: Well Im thinking something like this:
```
export type Region =
  | 'sierra-nevada'
  | 'southern-california'
  | 'central-coast'
  | 'northern-california';

export const Regions: Record<Region, string> = {
  'sierra-nevada': 'Sierra Nevada',
  'central-coast': 'Central Coast',
  'northern-california': 'Northern California',
  'southern-california': 'Southern California',
};

```

where the key is the actual value that the backend uses, and we have a record with the literal display string.  No regionToString function or extra stuff. But...that would require a major refactor

File: src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.ts [hunk 2]
```diff
@@ -131,8 +131,10 @@ export class FilterDropdownComponent<T> implements OnInit {
     this.previousSelections = this.selectedItems.slice();
   }
 
-  clearSelections(e: any): void {
+  clearSelections(e: Event): void {
```
Comment (lastminutediorama) [inline src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.ts:134]: 👍

**Review by lastminutediorama — APPROVED** [review]
lgtm.

**Pull request merged**